crates/api_models/src/revenue_recovery_data_backfill.rs

```diff
@@ -3,11 +3,49 @@ use std::{collections::HashMap, fs::File, io::BufReader};
 use actix_multipart::form::{tempfile::TempFile, MultipartForm};
 use actix_web::{HttpResponse, ResponseError};
 use common_enums::{CardNetwork, PaymentMethodType};
-use common_utils::events::ApiEventMetric;
+use common_utils::{events::ApiEventMetric, pii::PhoneNumberStrategy};
 use csv::Reader;
 use masking::Secret;
 use serde::{Deserialize, Serialize};
-use time::Date;
+use time::{Date, PrimitiveDateTime};
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum ScheduledAtUpdate {
+    SetToDateTime(#[serde(deserialize_with = "deserialize_primitive_datetime")] PrimitiveDateTime),
+    SetToNull(String), // matches "null" string
+}
```

Comment:
With this you won't be needing the custom deserializer

```suggestion
pub enum ScheduledAtUpdate {
    SetToDateTime(#[serde(with = "common_utils::custom_serde::iso8601")] PrimitiveDateTime),
    #[serde(rename = "null")]
    SetToNull, // matches "null" string
}
```